pandas/core/generic.py

```diff
@@ -8564,6 +8523,246 @@ def nanptp(values, axis=0, skipna=True):
+                A     B     C
+    2000-01-01   0.0   2.0   4.0
+    2000-01-02   3.0   5.0   7.0
+    2000-01-03   6.0   8.0  10.0
+    2000-01-04   NaN   NaN   NaN
+    2000-01-05   NaN   NaN   NaN
+    2000-01-06   NaN   NaN   NaN
+    2000-01-07   NaN   NaN   NaN
+    2000-01-08  21.0  23.0  25.0
+    2000-01-09  24.0  26.0  28.0
+    2000-01-10  27.0  29.0  31.0
+
+    >>> def g(s):
+    ...     "input is a Series whose index may be modified"
+    ...     i = np.arange(s.size)
+    ...     s.index = i
+    ...     return s + i
+
+    >>> df.transform(g)
+        A     B     C
+    0   0.0   1.0   2.0
+    1   4.0   5.0   6.0
+    2   8.0   9.0  10.0
+    3   NaN   NaN   NaN
+    4   NaN   NaN   NaN
+    5   NaN   NaN   NaN
+    6   NaN   NaN   NaN
+    7  28.0  29.0  30.0
+    8  32.0  33.0  34.0
+    9  36.0  37.0  38.0
+
+    Apply functions to specific columns by passing a dictionary mapping
+    column names to functions.
+
+    >>> def log_T(s):
+    ...     return np.log(10 + s)
+    >>> def normalize_T(s):
+    ...     return (s - s.mean()) / s.std()
+
+    >>> dic = {
+    ...     'A': lambda x: x+1,
+    ...     'B': lambda x: x+2,
+    ... }
+    >>> df.transform(dic)
+               A     B
+    2000-01-01   1.0   3.0
+    2000-01-02   4.0   6.0
+    2000-01-03   7.0   9.0
+    2000-01-04   NaN   NaN
+    2000-01-05   NaN   NaN
+    2000-01-06   NaN   NaN
+    2000-01-07   NaN   NaN
+    2000-01-08  22.0  24.0
+    2000-01-09  25.0  27.0
+    2000-01-10  28.0  30.0
+
+    When multiple functions are applied to a column, the result is a MultiIndex
+    in the columns.
+
+    >>> dic = {'B': [log_T, normalize_T]}
+    >>> df.transform(dic)
+                                   B
+                   log_T normalize_T
+    2000-01-01  2.397895   -1.143001
+    2000-01-02  2.639057   -0.889001
+    2000-01-03  2.833213   -0.635001
+    2000-01-04       NaN         NaN
+    2000-01-05       NaN         NaN
+    2000-01-06       NaN         NaN
+    2000-01-07       NaN         NaN
+    2000-01-08  3.465736    0.635001
+    2000-01-09  3.555348    0.889001
+    2000-01-10  3.637586    1.143001
+
+    >>> df[['A', 'B']].transform([log_T, normalize_T])
+                                   A                     B
+                   log_T normalize_T     log_T normalize_T
+    2000-01-01  2.302585   -1.143001  2.397895   -1.143001
+    2000-01-02  2.564949   -0.889001  2.639057   -0.889001
+    2000-01-03  2.772589   -0.635001  2.833213   -0.635001
+    2000-01-04       NaN         NaN       NaN         NaN
+    2000-01-05       NaN         NaN       NaN         NaN
+    2000-01-06       NaN         NaN       NaN         NaN
+    2000-01-07       NaN         NaN       NaN         NaN
+    2000-01-08  3.433987    0.635001  3.465736    0.635001
+    2000-01-09  3.526361    0.889001  3.555348    0.889001
+    2000-01-10  3.610918    1.143001  3.637586    1.143001
+
+    >>> s = pd.Series(np.sin([3*x for x in range(10)]), name='A',
```

Comment:
Again, I think a shorter example would demonstrate the point better.

Reply:
The reason I did not use random is that the validation script does not pass (random seed issue I guess). So I wanted 'random looking' values hence the sin().  
Anyway I simplified (a bit) and shortened the series.